syft/pkg/cataloger/golang/scan_binary.go

```diff
@@ -30,14 +37,42 @@ func scanFile(reader unionreader.UnionReader, filename string) ([]*debug.BuildIn
 		if bi == nil {
 			continue
 		}
-		builds = append(builds, bi)
+
+		v, err := getCryptoInformation(r)
+		if err != nil {
+			log.WithFields("file", filename, "error", err).Trace("unable to read golang version info")
+			continue
+		}
+
+		builds = append(builds, &ExtendedBuildInfo{bi, v})
 	}
 
 	archs := getArchs(readers, builds)
 
 	return builds, archs
 }
 
+func getCryptoInformation(reader io.ReaderAt) (string, error) {
+	v, err := version.ReadExeFromReader(reader)
+	if err != nil {
+		return "", err
+	}
+
+	cryptoInfo := ""
```

Comment:
it would be nice to have this kind of extraction logic into it's own function so that can be unit tested (given a `version.Version` instance assert that the correct []string value is returned)